olp-cpp-sdk-core/src/network2/NetworkCurl.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "olp/core/network2/NetworkCurl.h"
+#include "olp/core/network2/NetworkTypes.h"
+
+#include <atomic>
+
+#include "olp/core/logging/Log.h"
+
+#include <curl/curl.h>
+
+namespace olp {
+namespace network2 {
+
+namespace {
+constexpr auto kLogTag = "NetworkCurl";
+
+class NetworkCurlImpl final {
+ public:
+  RequestId Send(NetworkRequest request,
+                 std::shared_ptr<std::ostream> /*payload*/,
+                 Network::Callback /*callback*/,
+                 Network::HeaderCallback /*header_callback*/,
+                 Network::DataCallback /*data_callback*/) noexcept {
+    RequestId currentId = id_counter_++;
```

Comment:
With using post-increment the first id is invalid one
I dont think this is what you want.
Also please watch out for code style:
```suggestion
    RequestId current_id = id_counter_++;
```